File: Java-Core/src/main/java/presto/expression/SelectorExpression.java
package presto.expression;

import presto.error.SyntaxError;
import presto.grammar.UnresolvedIdentifier;
import presto.runtime.Context;
import presto.type.IType;


public abstract class SelectorExpression implements IExpression {

	IExpression parent;
	
	public SelectorExpression() {
	}
	
	public SelectorExpression(IExpression parent) {
		this.parent = parent;
	}
	
	public IExpression getParent() {
		return parent;
	}
	
	public void setParent(IExpression parent) {
		this.parent = parent;
	}
	
	public IType checkParent(Context context) throws SyntaxError {
		if (parent instanceof UnresolvedIdentifier)
			return ((UnresolvedIdentifier)parent).checkMember(context);
		else
			return parent.check(context);
 	}


}

File: Java-CodeServer/src/main/resources/js/prompto/expression/SelectorExpression.js
var UnresolvedIdentifier;

exports.resolve = function() {
    UnresolvedIdentifier = require("../grammar/UnresolvedIdentifier").UnresolvedIdentifier;
}

function SelectorExpression(parent) {
	this.parent = parent || null;
	return this;
}

SelectorExpression.prototype.checkParent = function(context) {
    if (this.parent instanceof UnresolvedIdentifier)
        return this.parent.checkMember(context);
    else
        return this.parent.check(context);
};


exports.SelectorExpression = SelectorExpression;